packages/core/src/config.ts

```diff
@@ -391,16 +394,93 @@ export async function loadConfig({
   }
 }
 
+export const getRsbuildInspectConfig = ({
+  normalizedConfig,
+  inspectOptions,
+  pluginManager,
+}: {
+  normalizedConfig: NormalizedConfig;
+  inspectOptions: InspectConfigOptions;
+  pluginManager: PluginManager;
+}): {
+  rawRsbuildConfig: string;
+  rsbuildConfig: InspectConfigResult['origin']['rsbuildConfig'];
+  rawEnvironmentConfigs: Array<{
+    name: string;
+    content: string;
+  }>;
+  environmentConfigs: InspectConfigResult['origin']['environmentConfigs'];
+} => {
+  const { environments, ...rsbuildConfig } = normalizedConfig;
+
+  const pluginNames = pluginManager.getPlugins().map((p) => p.name);
+
+  const rsbuildDebugConfig: Omit<NormalizedConfig, 'environments'> & {
+    pluginNames: string[];
+  } = {
+    ...rsbuildConfig,
+    pluginNames,
+  };
+
+  const rawRsbuildConfig = stringifyConfig(
+    rsbuildDebugConfig,
+    inspectOptions.verbose,
+  );
+
+  const { environmentDebugConfigs, rawEnvironmentConfigs } = Object.entries(
```

Comment:
I think we do not need to use `Object.entries().reduce()` loop here.

Just declare the variables and assign to it, the code will be much simpler.

- before:

```ts
const { environmentDebugConfigs, rawEnvironmentConfigs } = Object.entries(
  environments,
).reduce<{
  environmentDebugConfigs: Record<
    string,
    NormalizedEnvironmentConfig & {
      pluginNames: string[];
    }
  >;
  rawEnvironmentConfigs: Array<{
    name: string;
    content: string;
  }>;
}>(
  (prev, [name, config]) => {
    const debugConfig = {
      ...config,
      pluginNames,
    };
    prev.rawEnvironmentConfigs.push({
      name,
      content: stringifyConfig(debugConfig, inspectOptions.verbose),
    });
    prev.environmentDebugConfigs[name] = debugConfig;
    return prev;
  },
  {
    environmentDebugConfigs: {},
    rawEnvironmentConfigs: [],
  },
);
```

- after:

```ts
const environmentConfigs: Record<
  string,
  NormalizedEnvironmentConfig & { pluginNames: string[] }
> = {};

const rawEnvironmentConfigs: Array<{ name: string; content: string }> = [];

Object.entries(environments).forEach(([name, config]) => {
  const debugConfig = {
    ...config,
    pluginNames,
  };
  prev.rawEnvironmentConfigs.push({
    name,
    content: stringifyConfig(debugConfig, inspectOptions.verbose),
  });
  prev.environmentConfigs[name] = debugConfig;
});
```